src/openforms/formio/components/vanilla.py

```diff
@@ -306,11 +349,49 @@ def localize(self, component: SelectComponent, language_code: str, enabled: bool
             return
         translate_options(options, language_code, enabled)
 
+    def build_serializer_field(
+        self, component: SelectComponent
+    ) -> serializers.MultipleChoiceField:
+        validate = component.get("validate", {})
+        required = validate.get("required", False)
+        choices = [
+            (value["value"], value["label"]) for value in component["data"]["values"]
+        ]
+        return serializers.MultipleChoiceField(
+            choices=choices,
+            required=required,
+            allow_blank=not required,
+            allow_null=not required,
+        )
+
 
 @register("currency")
-class Currency(BasePlugin):
+class Currency(BasePlugin[CurrencyComponent]):
     formatter = CurrencyFormatter
 
+    def build_serializer_field(
+        self, component: CurrencyComponent
+    ) -> serializers.FloatField:
+        validate = component.get("validate", {})
+        required = validate.get("required", False)
+
+        extra = {}
+        if max_value := validate.get("max"):
+            extra["max_value"] = max_value
+        if min_value := validate.get("min"):
+            extra["min_value"] = min_value
```

Comment:
I think that this will not have the expected result if `max_value` or `min_value` are 0?